scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/NovaseqXProcessor.java

```diff
@@ -0,0 +1,241 @@
+package ca.on.oicr.gsi.runscanner.scanner.processor;
+
+import ca.on.oicr.gsi.runscanner.dto.IlluminaDragenNotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.IlluminaNotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.NotificationDto;
+import ca.on.oicr.gsi.runscanner.dto.type.AnalysisStatus;
+import ca.on.oicr.gsi.runscanner.dto.type.DRAGENWorkflow;
+import ca.on.oicr.gsi.runscanner.scanner.processor.NovaseqXProcessor.BCLConvertAnalysis.Analysis;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TimeZone;
+
+public class NovaseqXProcessor extends DefaultIllumina {
+  static class BCLConvertAnalysis {
```

Comment:
I had kept everything to one file to avoid an explosion of DRAGEN-related files in the processor package. I can change this if that's preferred though

Reply:
It's not going to be a lot of files, though it doesn't really matter if it is. You can make a sub-package to keep all the analysis/workflow classes in their own space